.github/workflows/test.yml

```diff
@@ -14,72 +13,30 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: "true"
+
       - name: Set up Python 3.11
         uses: actions/setup-python@v5
         id: setup_python
         with:
-          python-version: 3.11
-      - name: Restore cached virtualenv
-        id: restore_virtualenv_cache
-        uses: actions/cache/restore@v4
-        with:
-          key: venv-${{ runner.os }}-${{ steps.setup_python.outputs.python-version }}-${{ hashFiles('**/requirements.txt') }}
-          path: venv
-      - if: ${{ steps.restore_virtualenv_cache.outputs.cache-hit != 'true' }}
-        name: Cache pip dependencies
-        id: cache
-        uses: actions/cache@v4
-        env:
-          cache-name: cache-python-dependencies
-        with:
-          path: ~\AppData\Local\pip\Cache
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/requirements.txt') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - if: ${{ steps.restore_virtualenv_cache.outputs.cache-hit != 'true' }}
-        name: Create virtualenv if not exists
-        shell: bash
-        run: |
-          python -m venv venv
-      - if: ${{ steps.restore_virtualenv_cache.outputs.cache-hit != 'true' }}
-        name: Install dependencies
-        shell: bash
-        run: |
-          source venv/Scripts/activate
-          python -m pip install --upgrade pip
-          pip install wheel
-          pip install flake8
-          pip install pytest-cov
-          pip install -r requirements.txt
-          pip install -e parsley
-          find sinks/ sources/ tools/ -name requirements.txt -type f -exec pip install -r {} \;
-      - if: ${{ steps.restore_virtualenv_cache.outputs.cache-hit != 'true' }}
-        name: Saved cached virtualenv
-        uses: actions/cache/save@v4
-        with:
-          key: venv-${{ runner.os }}-${{ steps.setup_python.outputs.python-version }}-${{ hashFiles('**/requirements.txt') }}
-          path: venv
```

Comment:
not blocking, just wanted to say it is indeed probably good to just remove the caching since it probably doesn't have that much benefit